lib/pure/terminal.nim

```diff
@@ -267,6 +272,47 @@ else:
     mode.c_cc[VTIME] = 0.cuchar
     discard fd.tcSetAttr(time, addr mode)
 
+  proc getCursorPos*(): tuple [x, y: int] {.raises: [ValueError, IOError].} =
+    ## Returns cursor position (x, y)
+    var
+      xStr = ""
+      yStr = ""
+      ch: char
+      ct: int
+      readX = false
+
+    # use raw mode to ask terminal for cursor position
+    let fd = getFileHandle(stdin)
+    var oldMode: Termios
+    discard fd.tcGetAttr(addr oldMode)
+    fd.setRaw()
+    stdout.write(getPos)
```

Comment:
Oof, I just realized that it doesn't behave very well if there's not a terminal in the loop (writes the request to stdout and blocks while waiting for something from stdin).  It might do to warn users about that.  Added a comment.